receiver/splunkhecreceiver/splunkhec_to_metricdata.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhecreceiver
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/splunk"
+)
+
+// SplunkHecToMetricsData converts Splunk HEC metric points to
+// consumerdata.MetricsData. Returning the converted data and the number of
+// dropped time series.
+func SplunkHecToMetricsData(logger *zap.Logger, events []*splunk.Event, resourceCustomizer func(pdata.Resource)) (pdata.Metrics, int) {
+
+	numDroppedTimeSeries := 0
+	md := pdata.NewMetrics()
+
+	for _, event := range events {
+		resourceMetrics := pdata.NewResourceMetrics()
+		resourceMetrics.InitEmpty()
+
+		metrics := pdata.NewInstrumentationLibraryMetrics()
+		metrics.InitEmpty()
+		resourceMetrics.Resource().InitEmpty()
+		attrs := resourceMetrics.Resource().Attributes()
+		attrs.InitEmptyWithCapacity(3)
+		attrs.InsertString(conventions.AttributeHostHostname, event.Host)
+		attrs.InsertString(conventions.AttributeServiceName, event.Source)
+		attrs.InsertString(splunk.SourcetypeLabel, event.SourceType)
+		resourceCustomizer(resourceMetrics.Resource())
+
+		values := event.GetMetricValues()
+
+		labelKeys, labelValues := buildLabelKeysAndValues(event.Fields)
+		populateLabels := func(labels pdata.StringMap) {
+			for i, k := range labelKeys {
+				labels.Insert(k, labelValues[i])
+			}
+		}
+
+		metricNames := make([]string, 0, len(values))
+		for k := range values {
+			metricNames = append(metricNames, k)
+		}
+		sort.Strings(metricNames)
+
+		for _, metricName := range metricNames {
+			pointTimestamp := convertTimestamp(event.Time)
+			metric := pdata.NewMetric()
+			metric.InitEmpty()
+			metric.SetDataType(pdata.MetricDataTypeNone)
+			metric.SetName(metricName)
+
+			metricValue := values[metricName]
+			if i, ok := metricValue.(int64); ok {
+				addIntGauge(pointTimestamp, i, metric, populateLabels)
+			} else if i, ok := metricValue.(*int64); ok {
+				addIntGauge(pointTimestamp, *i, metric, populateLabels)
+			} else if f, ok := metricValue.(float64); ok {
+				if f == float64(int64(f)) {
+					addIntGauge(pointTimestamp, int64(f), metric, populateLabels)
```

Comment:
This still concerns me.  If a metric value happens to fall on an integer some times and other times is not, this could cause the metric type to flip-flop between intgauge and doublegauge, which doesn't seem good.

Reply:
Just make it always double gauge unless there is some really good reason specific to HEC that it shouldn't.

Reply:
Sounds good. Removed that code.